File: typescript/tests/rottingOranges/rottingOranges.test.ts
import { orangesRotting } from "../../src/rottingOranges/rottingOranges";

describe('test orangesRotting function', () => {
    it('should calculate the number of minutes to rot all oranges', () => {
        const grid = [[2, 1, 1], [1, 1, 0], [0, 1, 1]];
        const expected = 4;
        const actual = orangesRotting(grid);
        expect(expected).toEqual(actual);
    });

    it('should retrun -1 if all the oranges canot become rotten', () => {
        const grid = [[2, 1, 1], [0, 1, 1], [1, 0, 1]];
        const expected = -1;
        const actual = orangesRotting(grid);
        expect(expected).toEqual(actual);
    });
});

File: python/tests/rotting_oranges/rotting_oranges_test.py
import unittest

from rotting_oranges.rotting_oranges import Solution


class TestSolution(unittest.TestCase):
    def test_Rotting_Oranges(self):
        solution = Solution()
        grid = [[2, 1, 1], [1, 1, 0], [0, 1, 1]]
        expected = 4
        actual = solution.orangesRotting(grid)
        self.assertEqual(expected, actual)

    def test_Rotting_Oranges_Two(self):
        solution = Solution()
        grid = [[2, 1, 1], [0, 1, 1], [1, 0, 1]]
        expected = -1
        actual = solution.orangesRotting(grid)
        self.assertEqual(expected, actual)


